### PR TITLE
 Refactor/shrink side effect

### LINKED ISSUE #863: enhancement the use of Shrinking Bundle Size for solve sideEffect

### PR DESCRIPTION
**Pull request by xiaoluoHe**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [x] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #863 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/core/vchart.ts [hunk 1]
```diff
@@ -82,49 +82,71 @@ import {
...
+  static useRegisters(comps: (() => void)[]) {
+    comps.forEach((fn: () => void) => {
+      fn();
+    });
+  }
+
   /**
    * 注册图表
    * @param charts 图表类
+   * @deprecated
```
Comment (kkxxkk2019) [inline packages/vchart/src/core/vchart.ts:117]: 这里可以补充一下注释，用什么方法替代，给个新方法的例子

File: packages/vchart/src/core/vchart.ts [hunk 2]
```diff
@@ -1578,3 +1602,22 @@ export class VChart implements IVChart {
...
+  // install region module
+  Factory.registerRegion('region', Region);
+  // install layout module
+  Factory.registerLayout('base', Layout);
+  // install essential marks
+  Factory.registerMark(GroupMark.type, GroupMark);
+  // install essential vgrammar transform
+  View.useRegisters([registerFilterTransform, registerMapTransform]);
+  // install animation
+  registerVGrammarAnimation();
```
Comment (kkxxkk2019) [inline packages/vchart/src/core/vchart.ts:1616]: 动画的注册可以再拆分去吗？

**Comment by kkxxkk2019** [comment]
记得补充文档

File: packages/vchart/src/core/vchart.ts (hunk 1 above)
Comment (xiaoluoHe) [reply packages/vchart/src/core/vchart.ts:117]: 想了下，这里还是不标 deprecated 。
后续扩展不存在的 chart/mark/layout 的时候，还是要用这个 API

File: packages/vchart/src/core/vchart.ts (hunk 2 above)
Comment (xiaoluoHe) [reply packages/vchart/src/core/vchart.ts:1616]: 倒是可以拆到图元注册里

File: packages/vchart/src/core/vchart.ts (hunk 1 above)
Comment (kkxxkk2019) [reply packages/vchart/src/core/vchart.ts:117]: OK

**Review by kkxxkk2019 — APPROVED** [review]

**Pull request merged**